wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/server/HttpBitbucketServerApiClient.java

```diff
@@ -420,4 +425,11 @@ private ScmUnauthorizedException buildScmUnauthorizedException() {
         "1.0",
         authenticator.getLocalAuthenticateUrl());
   }
+
+  private void assertAuthenticatorNonNull() throws ScmCommunicationException {
```

Comment:
Why don't we check that in class costructor?

Reply:
reworked to use Noop Authenticator